.github/workflows/tests.yml

```diff
@@ -48,13 +48,46 @@ jobs:
           toolchain: ${{ matrix.rust-version }}
           override: true
 
-      - name: Install libpq (Windows)
-        if: matrix.os == 'windows-latest'
+      - name: Install PostgreSQL and create a database  (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update && sudo apt-get install -y libpq-dev postgresql
+          sudo service postgresql start
+          sudo -u postgres createuser -s devel
+          sudo -u postgres createdb -O devel devel
+          sudo -u postgres psql -c "ALTER USER devel PASSWORD 'devel';" devel
+          echo '::set-env name=DATABASE_URL::postgres://devel:devel@localhost/devel'
+
+      - name: Install PostgreSQL and create a database (MacOS)
+        if: runner.os == 'macOS'
+        run: |
+          /usr/local/opt/postgres/bin/pg_ctl -D /usr/local/var/postgres start
+          /usr/local/opt/postgres/bin/createuser -s devel
+          /usr/local/opt/postgres/bin/createdb -O devel devel
+          /usr/local/opt/postgres/bin/psql -c "ALTER USER devel PASSWORD 'devel';" devel
+          echo '::set-env name=DATABASE_URL::postgres://devel:devel@localhost/devel'
+
+      - name: Install PostgreSQL and create a database (Windows)
+        if: runner.os == 'Windows'
         shell: bash
         run: |
-          choco install postgresql12 --force --params '/Password:root'
+          choco install postgresql12 --force
           echo '::add-path::C:\Program Files\PostgreSQL\12\bin'
           echo '::add-path::C:\Program Files\PostgreSQL\12\lib'
+          echo '::set-env name=PQ_LIB_DIR::C:\Program Files\PostgreSQL\12\lib'
+          "C:\Program Files\PostgreSQL\12\bin\createuser" -s devel
+          "C:\Program Files\PostgreSQL\12\bin\createdb" -O devel devel
+          "C:\Program Files\PostgreSQL\12\bin\psql" -c "ALTER USER devel PASSWORD 'devel';" devel
+          echo '::set-env name=DATABASE_URL::postgres://devel:devel@localhost/devel'
```

Comment:
**Q:** I wonder, would it make sense to consider using `services` feature [1] to spin up a postgresql container and avoid duplicate commands for every OS we support?  

[1] https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-postgresql-service-containers#running-jobs-directly-on-the-runner-machine

Reply:
As discussed, that's only possible on Linux at the moment :( The docs you pointed me to contained a valuable hint, though: PostgreSQL is already installed on Windows; it's just disabled. I will try to update this script to start the service instead of reinstalling it from scratch, which should shave off some running time.